tests/php/sync/test_class.jetpack-sync-full.php

```diff
@@ -307,6 +307,64 @@ function test_full_sync_sends_all_term_relationships() {
 		$this->assertEquals( $original_number_of_term_relationships, $replica_number_of_term_relationships );
 	}
 
+	function test_full_sync_enqueue_term_relationships() {
+		global $wpdb;
+
+		// how many items are we allowed to enqueue on a single request/continue_enqueuing
+		$max_enqueue_full_sync = 2;
+		// how many items are we allowed to enqueue on any given time
+		$max_queue_size_full_sync = 3;
+		// how many term relationships we can put on a full_sync_term_relationships item
+		$sync_item_size = 4;
+
+		Settings::update_settings( [
+			'term_relationships_full_sync_item_size' => $sync_item_size,
+			'max_queue_size_full_sync'               => $max_queue_size_full_sync,
+			'max_enqueue_full_sync'                  => $max_enqueue_full_sync,
+		] );
+
+		$post_ids = $this->factory->post->create_many( 4 );
+
+		foreach ( $post_ids as $post_id ) {
+			wp_set_object_terms( $post_id, array( 'cat1', 'cat2', 'cat3' ), 'category', true );
+			wp_set_object_terms( $post_id, array( 'tag1', 'tag2', 'tag3' ), 'post_tag', true );
+		}
+
+		// 28
+		$original_number_of_term_relationships = $wpdb->get_var( "SELECT COUNT(*) FROM $wpdb->term_relationships" );
+		// ceil(28/4) = 7
+		$total_items = intval( ceil( $original_number_of_term_relationships  / $sync_item_size ) );
+
+		$this->full_sync->start( array( 'term_relationships' => true ) );
+		$this->sender->do_full_sync(); // empty the queue since – "full_sync_start" takes one item in the queue
+
+		$status = $this->full_sync->get_enqueue_status();
+		list( $total, $initial_queued, $finished ) = $status['term_relationships'];
+
+		$this->assertEquals( $total_items, $total );
+		$this->assertEquals( $max_enqueue_full_sync, $initial_queued );
+		$this->assertNotTrue( $finished );
+
+		$this->full_sync->continue_enqueuing(); // try to enqueue $max_enqueue_full_sync items
+		$this->full_sync->continue_enqueuing(); // try to enqueue $max_enqueue_full_sync items
+
+		// hit $max_queue_size_full_sync limit
+		$status = $this->full_sync->get_enqueue_status();
+		list( $total, $queued, $finished ) = $status['term_relationships'];
```

Comment:
Let's add a check that we're not finished:

```suggestion
		list( $total, $queued, $finished ) = $status['term_relationships'];
		$this->assertNotTrue( $finished );
```